src/wasm/encoder.cc

```diff
@@ -58,13 +58,19 @@ struct WasmFunctionBuilder::Type {
   LocalType type_;
 };
 
-WasmFunctionBuilder::WasmFunctionBuilder(Zone* zone)
+WasmFunctionBuilder::WasmFunctionBuilder(Zone* zone, const std::string& name)
```

Comment:
Are we gaining much by using std::string here? AFAICT we build up the std string with a quite slow manual memory copy, just to do a memcpy of the bytes into the module later.